### DISCUSSION TITLE
Flash Circular Buffer (FCB) for Large Data Logging - Sample Code Needed

### DISCUSSION THREAD

**Discussion by Ananyaaynana** [zephyrproject-rtos/zephyr#74984] [Q&A]
I'm investigating methods for logging large amounts of data to flash memory in a Zephyr project.

I initially attempted using the NVS (Non-Volatile Storage) API for logging large amounts of data into the flash. However, I encountered continuous garbage collection, which caused performance issues.

I'm interested in exploring the Flash Circular Buffer (FCB) as an alternative solution for large data logging. Unfortunately, I haven't been able to locate sample code that demonstrates its implementation for this specific use case.

NVS Experience:

For reference, I previously used NVS, but it resulted in continuous garbage collection, hindering performance.

What would be the best method to log large amounts of data into the flash where the flash is treated as FIFO.

**Comment by ShahinHaque97-CT** [comment]
THE FCB ztests can be used as a sample replacement while waiting for a sample. There are tests for initialization, reading, writing and erasing.
